tensorflow/python/training/optimizer.py

```diff
@@ -282,8 +282,14 @@ def apply_gradients(self, grads_and_vars, global_step=None, name=None):
     # This is a default implementation of apply_gradients() that can be shared
     # by most optimizers.  It relies on the subclass implementing the following
     # methods: _create_slots(), _prepare(), _apply_dense(), and _apply_sparse().
+
+    grads_to_tensors_list = []
+    vars_list = []
     grads_and_vars = tuple(grads_and_vars)  # Make sure repeat iteration works
     for g, v in grads_and_vars:
+      # Check if gradient is Variable. If so, convert it to tensor
+      if isinstance(g, (variables.Variable)):
+        g = tf.convert_to_tensor(g)
```

Comment:
It seems like it would be simpler (and cover more cases) to do:

```python
if g is not None:
  try:
    g = ops.convert_to_tensor_or_indexed_slices(g)
  except TypeError:
    raise TypeError(
        "Gradient must be convertible to a Tensor or IndexedSlices, or None: %s" %g)
```